File: agri-ecommerce-backend/src/main/java/com/agri/ecommerce/service/AuthService.java
package com.agri.ecommerce.service;

import com.agri.ecommerce.dto.request.auth.ForgotPasswordRequest;
import com.agri.ecommerce.dto.request.auth.LoginRequest;
import com.agri.ecommerce.dto.request.auth.RegisterRequest;
import com.agri.ecommerce.dto.request.auth.ResetPasswordRequest;
import com.agri.ecommerce.dto.response.auth.AuthResponse;
import com.agri.ecommerce.dto.response.auth.PasswordResetResponse;
import com.agri.ecommerce.dto.response.user.UserResponse;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    PasswordResetResponse forgotPassword(ForgotPasswordRequest request);

    PasswordResetResponse resetPassword(ResetPasswordRequest request);

    UserResponse getCurrentUser(Long userId);
}

File: agri-ecommerce-frontend/src/services/auth.service.js
import axiosClient from "@/lib/axios-client";

export const authService = {
  register: async (payload) => {
    return axiosClient.post("/auth/register", payload);
  },

  login: async (payload) => {
    return axiosClient.post("/auth/login", payload);
  },

  me: async () => {
    return axiosClient.get("/auth/me");
  },

  forgotPassword: async (payload) => {
    return axiosClient.post("/auth/forgot-password", payload);
  },

  resetPassword: async (payload) => {
    return axiosClient.post("/auth/reset-password", payload);
  },
};
